lib/std/os/windows.zig

```diff
@@ -935,6 +936,22 @@ pub fn DeleteFile(sub_path_w: []const u16, options: DeleteFileOptions) DeleteFil
         .CANNOT_DELETE => return error.AccessDenied,
```

Comment:
Because this `NtCreateFile` call does not use `FILE_DELETE_ON_CLOSE` anymore, it seems likely that this error is no longer possible here but likely is possible from the `NtSetInformationFile` call.

More context:
- https://github.com/ziglang/zig/pull/10495
- https://github.com/ziglang/zig/issues/13084

It'd probably be good to add a test case that intentionally tries to delete a read-only file so that we know this case continues to be handled in the same way.

Reply:
You are right. It should be moved to the set function return. 
Also, while testing this, I found out that `NtCreateFile` can also result in AccessDenied, when the owner is different. So we can replace `CANNOT_DELETE` with `ACCESS_DENIED` and add `CANNOT_DELETE` to the second rc.

Reply:
Note on the test example:
I think you may have found another bug with that :smile: . The test will fail on Access Denied for a different reason. The file is not open with `FILE_WRITE_ATTRIBUTES` as the only `OpenMode` are `read_only` (GENERIC_READ), write_only (GENERIC_WRITE), and read_write (both). 

and since `GENERIC_READ` only have `FILE_READ_ATTRIBUTES`, it cannot change its attributes anymore after the read only is set.
I am looking to see if there are more refined flags or functions that allows that, but from a quick search, i couldn't find one.

Reply:
Looks like this is the relevant issue: https://github.com/ziglang/zig/issues/12377 (about `Dir` but seems like it might apply to File.OpenFlags too).

Reply:
Yep, that looks like the same behavior.
I am adding a modified test with no file closing/reopen. If it doesn't seem necessary to test right now due to this, I can remove it.